### PR TITLE
Add documentation about yielding declared scan profiles in normalizers

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
This updates some documentation pages, and in particular adds a section on the normalizers being allowed to yield `DeclaredScanProfile` objects.

### Issue link
Closes https://github.com/minvws/nl-kat-coordination/issues/2452

### Demo
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: docs/source/developer_documentation/boefjes.md [hunk 1]
```diff
@@ -112,25 +133,27 @@ subgraph Process 0
...
+  Worker[pid=1]->>Worker[pid=1]: receives SIGKILL
+  SchedulerRuntimeManager->>Worker[pid=1]: if not is_alive()
+  SchedulerRuntimeManager->>SharedDict: set p_item_id = SharedDict[1]
+  SchedulerRuntimeManager->>Scheduler: set p_item_id status to FAILED
+  SchedulerRuntimeManager->>Worker[pid=1]: close()
+  SchedulerRuntimeManager->>Worker[pid=2]: start()
 ```
 
-
+Here, the SharedDict is an object that keeps track of a mapping of the pid's of workers to the task they are handling.
```
Comment (ammar92) [inline docs/source/developer_documentation/boefjes.md:156]:
```suggestion
Here, the `SharedDict` serves as an object designed to maintain a mapping of workers' process IDs (PIDs) to the specific tasks they are currently handling.
```

File: docs/source/developer_documentation/boefjes.md [hunk 2]
```diff
@@ -234,28 +257,34 @@ Here a `Boefje` object that wraps this metadata is defined, as well as `Normaliz
...
-To lint the code using black, run:
+For the KATalogus integration tests, run:
+```shell
+$ make itest
+```
+
+To lint the code using pre-commit, run:
 ```shell
-$ python -m black .
+$ pre-commit run --all
```
Comment (ammar92) [inline docs/source/developer_documentation/boefjes.md:289]:
```suggestion
$ pre-commit run --all-files
```

File: docs/source/developer_documentation/rocky.md [hunk 3]
```diff
@@ -1,78 +1,127 @@
...
 ```
-ROCKY_DB_HOST=
-ROCKY_DB_PORT=
-ROCKY_DB=
-ROCKY_DB_USER=
-ROCKY_DB_PASSWORD=
-ROCKY_DB_DSN=
+
+#### Tip
+A local python environment is useful for unit testing even when using Docker.
```
Comment (ammar92) [inline docs/source/developer_documentation/rocky.md:107]:
```suggestion
A local Python environment is useful for unit testing even when using Docker.
```

File: docs/source/introduction/makeyourown.rst [hunk 4]
```diff
@@ -244,6 +250,45 @@ The file normalize.py contains the actual normalizer: Its only job is to parse r
...
+
+.. code-block:: python
+
+  def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI]:
+   ...
+   yield ip_address
+   yield DeclaredScanProfile(reference=ip_address.reference, level=2)
+
+
+Of course, when the normalizer id is not present in the whitelist, the yielded DeclaredScanProfiles are ignored.
```
Comment (ammar92) [inline docs/source/introduction/makeyourown.rst:289]:
```suggestion
Of course, when the normalizer id is not present in the whitelist, the yielded `DeclaredScanProfiles` are ignored.
```

File: docs/source/introduction/makeyourown.rst [hunk 5]
```diff
@@ -244,6 +250,45 @@ The file normalize.py contains the actual normalizer: Its only job is to parse r
                 yield ft
                 yield f
 
+Yielding a DeclaredScanProfile
+==============================
+
+Additionally, normalizers can yield DeclaredScanProfiles.
```
Comment (ammar92) [inline docs/source/introduction/makeyourown.rst:256]:
```suggestion
Additionally, normalizers can yield `DeclaredScanProfile`s.
```

File: docs/source/introduction/makeyourown.rst [hunk 6]
```diff
@@ -207,7 +212,8 @@ The normalizers translate the output of a boefje into objects that fit the data
 normalize.py
 ************
 
-The file normalize.py contains the actual normalizer: Its only job is to parse raw data and create, fill and yield the actual objects. (of valid object-types that are subclassed from OOI like IPPort)
+The file normalize.py contains the actual normalizer: Its only job is to parse raw data and create,
```
Comment (ammar92) [inline docs/source/introduction/makeyourown.rst:215]:
```suggestion
The file `normalize.py` contains the actual normalizer: Its only job is to parse raw data and create,
```

File: docs/source/introduction/makeyourown.rst [hunk 7]
```diff
@@ -188,7 +189,11 @@ The normalizer imports the raw information, extracts the objects from it and giv
 normalizer.json
 ***************
 
-The normalizers translate the output of a boefje into objects that fit the data model. Each normalizer defines what input it accepts and what object-types it provides. In the case of the shodan normalizer, it involves the entire output of the shodan boefje (created based on IP address), where findings and ports come out. The normalizer.json defines these:
+The normalizers translate the output of a boefje into objects that fit the data model.
+Each normalizer defines what input it accepts and what object-types it provides.
+In the case of the shodan normalizer,
+it involves the entire output of the shodan boefje (created based on IP address),
+where findings and ports come out. The normalizer.json defines these:
```
Comment (ammar92) [inline docs/source/introduction/makeyourown.rst:196]:
```suggestion
where findings and ports come out. The `normalizer.json` defines these:
```

**Review by ammar92 — COMMENTED** [review]
Looks good 👍 Just a few suggestions

File: docs/source/developer_documentation/rocky.md (hunk 3 above)
Comment (Donnype) [reply docs/source/developer_documentation/rocky.md:107]: I solved this in several other places as well :+1:

File: docs/source/introduction/makeyourown.rst (hunk 4 above)
Comment (Donnype) [reply docs/source/introduction/makeyourown.rst:289]: Not sure if the single quote is what we want, but I did some updates (see updated screenshot also)

File: docs/source/introduction/makeyourown.rst (hunk 5 above)
Comment (Donnype) [reply docs/source/introduction/makeyourown.rst:256]: Same here

**Review by ammar92 — DISMISSED** [review]

File: docs/source/introduction/makeyourown.rst [hunk 8]
```diff
@@ -179,7 +179,8 @@ The boefje itself imports the shodan api module, assigns an IP address to it and
 Normalizers
 -----------
 
-The normalizer imports the raw information, extracts the objects from it and gives them to Octopoes. Since OpenKAT 1.3.0, the normalizers are fully self-contained. They consist of the following files:
+The normalizer imports the raw information, extracts the objects from it and gives them to Octopoes.
+Since OpenKAT 1.3.0, the normalizers are fully self-contained. They consist of the following files:
```
Comment (dekkers) [inline docs/source/introduction/makeyourown.rst:183]: I know that it was already there, but given that we are making changes, I don't think it is relevant anymore that something was changed in OpenKAT 1.3.

```suggestion
The normalizer imports the raw information, extracts the objects from it and gives them to Octopoes. They consist of the following files:
```

**Review by underdarknl — APPROVED** [review]

**Pull request merged**